ui/src/main/java/edu/wpi/grip/ui/GRIPUIModule.java

```diff
@@ -0,0 +1,83 @@
+package edu.wpi.grip.ui;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.TypeLiteral;
+import com.google.inject.assistedinject.FactoryModuleBuilder;
+import com.google.inject.matcher.Matchers;
+import com.google.inject.spi.InjectionListener;
+import com.google.inject.spi.TypeEncounter;
+import com.google.inject.spi.TypeListener;
+import edu.wpi.grip.core.Source;
+import edu.wpi.grip.ui.annotations.ParametrizedController;
+import edu.wpi.grip.ui.components.StartStoppableButton;
+import edu.wpi.grip.ui.pipeline.OutputSocketController;
+import edu.wpi.grip.ui.pipeline.SocketHandleView;
+import edu.wpi.grip.ui.pipeline.StepController;
+import edu.wpi.grip.ui.pipeline.input.*;
+import edu.wpi.grip.ui.pipeline.source.CameraSourceController;
+import edu.wpi.grip.ui.pipeline.source.MultiImageFileSourceController;
+import edu.wpi.grip.ui.pipeline.source.SourceController;
+import javafx.fxml.FXMLLoader;
+
+import java.io.IOException;
+
+/**
+ * A Guice {@link com.google.inject.Module} for GRIP's UI package.
+ */
+public class GRIPUIModule extends AbstractModule {
+    @Override
+    protected void configure() {
+
+        bindListener(Matchers.any(), new TypeListener() {
+            @Override
+            public <I> void hear(final TypeLiteral<I> typeLiteral, TypeEncounter<I> typeEncounter) {
+                typeEncounter.register((InjectionListener<I>) i -> {
+                    if (!i.getClass().isAnnotationPresent(ParametrizedController.class)) {
+                        return;
+                    }
+                    try {
+                        FXMLLoader.load(i.getClass().getResource(
+                                i.getClass().getAnnotation(ParametrizedController.class).url()),
+                                null, null,
+                                c -> i
+                        );
```

Comment:
Perhaps move this into a utility so that tests can take advantage of it without having to use Guice.